fivekmrun_app_flutter/lib/state/locale_provider.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'dart:io';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class LocaleProvider extends ChangeNotifier {
+  Locale? _locale;
+
+  Locale? get locale => _locale;
+
+  LocaleProvider() {
+    _loadLocale();
+  }
+
+  Future<void> setLocale(Locale locale) async {
+    final preferences = await SharedPreferences.getInstance();
+    await preferences.setString('locale', locale.languageCode);
+    _locale = locale;
+    notifyListeners();
+  }
+
+  Future<void> _loadLocale() async {
+    final preferences = await SharedPreferences.getInstance();
+    final localeCode = preferences.getString('locale');
+
+    if (localeCode != null) {
+      _locale = Locale(localeCode);
+    } else {
+      Locale platformLocale = Locale(Platform.localeName);
```

Comment:
While testing on IOS I noticed that when I set the language and region to Bulgaria the `Platform.localeName` is actually `bg_BG` and not `bg` (the one defined in `AppLocalizations.supportedLocales`). Check the screenshot.
So this piece of logic always selects `en`. Do you think there is way to only check the preferred language and not the region?

<img width="406" alt="Screenshot 2024-11-29 at 23 34 46" src="https://github.com/user-attachments/assets/952ce594-0402-4c79-8a25-f8c967d84bd6">